### DISCUSSION TITLE
Vite's hot reload resets dark mode setting when using useDarkMode()

### DISCUSSION THREAD

**Discussion by joonashak** [juliencrn/usehooks-ts#427] [Help] [answered]
Recently discovered your `useDarkMode()` hook, and love it! I'm just running into an issue with Vite, where the dev server's hot reload/HMR resets the dark mode setting.

Repro steps:

1. Have dark mode as browser preference. Dark mode is displayed correctly.
2. Switch to light mode. Reload page to confirm that the preference is persisted.
3. Save a file watched by Vite's dev server. Theme changes back to dark mode as the hot-reload happens in browser.

This is pretty frustrating when working on light mode's colors, etc. Is there a way to have the user preference persisted over hot-reloads? Thanks!

**Comment by juliencrn** [comment, marked as answer]
Hi @joonashak, thanks for this complete bug report, I will work on it next week.
I close the discussion since I've created a bug issue (#433) from this thread
